src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORKING_HOURS_INDEX = 2;
+    private static final int INCOME_PER_HOUR_INDEX = 3;
+    private static final String DATA_SEPARATOR = " ";
+    private static final String DATE_PATTERN = "dd.MM.yyyy";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaries = new int[names.length];
+        for (String line : data) {
```

Comment:
you should use only two for loops in you solution, you have three, you can go through list of names and inside of this for loop use inner for loop that will be going through lines of data to calculate salary for concrete name and add its salary to builder after this inner for loop works